src/js/components/List/List.js

```diff
@@ -349,6 +360,25 @@ const List = React.forwardRef(
                 let content;
                 let boxProps = {};
 
+                const key = getValue(item, index, itemKey) || index;
```

Comment:
@halocline Just FYI for review. This block of logic was lifted up from where it previously was on line 395 and modified to support the "object" type.

Just calling attention that much of this logic was existing and we're simply moving the location.